demo/src/bundles/demo/scroll-service.ts

```diff
@@ -5,41 +5,51 @@ const defaultDuration = 500; //ms
 @Injectable()
 export class ScrollService {
   public scrollCompleted = new EventEmitter();
-  private scrollContainer: HTMLElement;
-  private startTime: number;
-  private duration: number;
-  private startOffset: number;
-  private endOffset: number;
-  private raf: number;
-
-  public smoothScroll(scrollContainer: HTMLElement, endOffset: number, duration?: number) {
-    cancelAnimationFrame(this.raf);
+  private scrollContainer?: HTMLElement | Window;
+  private startTime?: number;
+  private duration?: number;
+  private startOffset?: number;
+  private endOffset?: number;
+  private raf?: number;
+
+  public smoothScroll(scrollContainer: HTMLElement | Window = window, endOffset: number, duration?: number) {
+    if (this.raf) {
+      cancelAnimationFrame(this.raf);
+    }
     this.duration = duration || defaultDuration;
-    this.scrollContainer = scrollContainer || <any>window;
-    this.startOffset = scrollContainer.scrollTop || window.pageYOffset;
+    this.scrollContainer = scrollContainer;
+    this.startOffset = scrollContainer instanceof Window ? scrollContainer.scrollY : scrollContainer.scrollTop;
```

Comment:
Just looking at the change, should it be `window.pageYOffset` instead of `scrollContainer.scrollY`?

Reply:
Not entirely sure about the code here but I'm fine as long as you've tested it